### PR TITLE
Http xml memory fix

### PR DESCRIPTION
**Pull request by nboard**
I fixed the major memory leak found in the HTTP driver, and also a minor memory leak found in the expression logic. 
I also included changes the XML driver converting the cache  from a hash-map to a hash-queue. Although this had far less effect on the memory usage than the HTTP driver's memory leak, I did notice that the memory usage now does not grow noticeably, whereas with just the leak fix it would still climb slowly.

### CODE REVIEW

**Comment by nboard** [comment]
I caught a mistake in the XML driver changes, and fixed another minor leak in the expression functions.

**Comment by nboard** [comment]
I noticed it was behind master. There were no conflicts, but I went ahead and merged master into it

File: centrallix/osdrivers/objdrv_xml.c [hunk 1]
```diff
@@ -214,7 +214,7 @@ typedef struct
 /*** GLOBALS ***/
 struct
     {
-    XHashTable	cache;
+    XHashQueue	cache;
```
Comment (gbeeley) [inline centrallix/osdrivers/objdrv_xml.c:217]: Should also #include "cxlib/xhashqueue.h" with the other includes at the beginning of the file, mainly for good hygiene.  The file is already included from obj.h, which is why things are working.

**Review by gbeeley — CHANGES_REQUESTED** [review]
One minor fix needed - thanks!

File: centrallix/osdrivers/objdrv_xml.c (hunk 1 above)
Comment (nboard) [reply centrallix/osdrivers/objdrv_xml.c:218]: Noted

**Review by gbeeley — APPROVED** [review]
Looks good - thanks!

**Pull request merged**